woo-razorpay.php

```diff
@@ -2073,3 +2199,76 @@ function disable_coupon_field_on_cart($enabled)
     add_action('woocommerce_cart_updated', 'enqueueScriptsFor1cc', 10);
     add_filter('woocommerce_order_needs_shipping_address', '__return_true');
 }
+
+// instrumentation
+
+// plugin activation hook
+function razorpayPluginActivated()
+{
+    $paymentSettings = get_option('woocommerce_razorpay_settings');
+
+    $trackObject  = new TrackPluginInstrumentation($paymentSettings['key_id'], $paymentSettings['key_secret']);
+
+    $activateProperties = [
+        'event_timestamp'     => time(),
+        'page_url'            => $_SERVER['HTTP_REFERER'],
+        'unique_id'           => $_SERVER['HTTP_HOST'],
+        'redirect_to_page'    => $_SERVER['REQUEST_SCHEME'] . '://' . $_SERVER['HTTP_HOST'].$_SERVER['REQUEST_URI']
+    ];
+
+    $response = $trackObject->rzpTrackSegment('plugin activate', $activateProperties);
+}
+
+// plugin deactivation hook
+function razorpayPluginDeactivated()
+{
+    $paymentSettings = get_option('woocommerce_razorpay_settings');
+
+    $trackObject  = new TrackPluginInstrumentation($paymentSettings['key_id'], $paymentSettings['key_secret']);
+
+    $api = new Api($paymentSettings['key_id'], $paymentSettings['key_secret']);
+
+    $orderCount = $api->request->request('GET', 'orders')['count'];
```

Comment:
@abdulwahidsharief , why we are fetching the order count here ? if the order already created by another plugin or manually?

Reply:
@ramth05 have to check if merchant is a transacting user